uber/models.py

```diff
@@ -473,11 +473,20 @@ def jobs_for_signups(self):
             return [job.to_dict(fields) for job in jobs if job.restricted or frozenset(job.hours) not in restricted_hours]
 
         def guess_attendee_watchentry(self, attendee):
-            return self.query(WatchList).filter(and_(or_(WatchList.first_names.contains(attendee.first_name),
-                                                         and_(WatchList.email != '', WatchList.email == attendee.email),
-                                                         and_(WatchList.birthdate != None, WatchList.birthdate == attendee.birthdate)),
-                                                     WatchList.last_name == attendee.last_name,
-                                                     WatchList.active == True)).all()
+            return self.query(WatchList).filter(
+                    and_(
+                        or_(
+                            WatchList.first_names.contains(attendee.first_name),
+                            and_(WatchList.email != '', WatchList.email == attendee.email),
+                            *[WatchList.birthdate == attendee.birthdate] if
+                                WatchList.birthdate and WatchList.birthdate != '' and
+                                attendee.birthdate and attendee.birthdate != ''
+                            else []
```

Comment:
this syntax is a bit weird to look at at first, but what it's doing is like the following:

ok_to_check_for_birthdate = WatchList.birthdate and WatchList.birthdate != '' and attendee.birthdate and attendee.birthdate != ''

if ok_to_check_for_birthdate:
   include this check in SQL: WatchList.birthdate == attendee.birthdate
if not, don't include it.